### PR TITLE
feat(usecase): new metadata to indenty the usecase

### PR DESCRIPTION
**Pull request by endersoncosta**
feature #51

## Proposed Changes

1. new metadata to indentify the usecase
2. new metadata to document the authorize request

### CODE REVIEW

**Comment by jhomarolo** [comment]
@endersoncosta before merge, could you please update the docs? And also the website.

**Comment by endersoncosta** [comment]
> 
> 
> @endersoncosta before merge, could you please update the docs? And also the website.

Of course.

**Comment by endersoncosta** [comment]
Documentation improved

https://github.com/herbsjs/herbsjs.github.io/pull/69

File: README.md [hunk 1]
```diff
@@ -53,13 +57,19 @@ const dependency = {
...
+        indentifier: "upsertItem",
 
         // Input/Request type validation
         request: { listId: Number, item: Item },
 
         // Output/Response type
         response: { item: Item },
 
+        // Authorization request
+        authorizeRequest: { user: User }
```
[the remark this answers is not in the document]
Comment (endersoncosta) [reply README.md:71]: We need the authorize request metadata to know what is necessary to run the authorization...

We can exemplify with the request of the useCase, there has what we need to call the usecase... but the authorize can recept anything...

then we need some data to identify what is necessary...

[the remark this answers is not in the document]
Comment (endersoncosta) [reply README.md:71]: the name of the metadata, we can change, but we still need of this information...

File: test/helpers/stringToCamelCase.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+const assert = require('assert')
+
+const { Ok } = require('../../src/results')
+const stringToCamelCase = require('../../src/helpers/stringToCamelCase')
```
[the remark this answers is not in the document]
Comment (endersoncosta) [reply test/helpers/stringToCamelCase.js:4]: I think the suma is a just a validator, I don't see it on there...

[the remark this answers is not in the document]
Comment (jhomarolo) [reply test/helpers/stringToCamelCase.js:4]: I see your point. But some libraries (herbs2knex, herbs2gql, etc), seems to use the same or similar functions using this "helper".

We could bring to suma those functions, beyond checks and validations... I don't know for sure, was just a brainstorm

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dalssoft) [reply README.md:71]: I think this should go to a different issue/PR.

**Comment by dalssoft** [comment]
As far as I'm concerned, the idea to have a unique identifier to an use case is to have a way to load UCs in different places like glues and CLI generated code. 

However, there is another discussion going on with a different approach but with the same goal (https://github.com/herbsjs/herbs/issues/11). They might even be complementary. But either way, I think we should evolve the discussion on the mentioned issue in order to have more info about the way we should take to solve this problem.

**Comment by jhomarolo** [comment]
>However, there is another discussion going on with a different approach but with the same goal (herbsjs/herbs#11). 

I see this PR, as a feature in the herbarium, like a new field to implement there, but of course, both are correlated

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhomarolo) [reply README.md]: There are any pendencies with this PR? @dalssoft, @endersoncosta

**Comment by jhomarolo** [comment]
@endersoncosta this pr is under conflict

**Comment by dalssoft** [comment]
With Herbarium I think this PR is not a priority (`new metadata to identify the use case`) since we should not have two concurrent methods for the same goal, at least not now.

Regarding `new metadata to document the authorize request`, I think we should create a new issue to discuss the problem and the possible solutions.

**Comment by italojs** [comment]
@dalssoft what do you think about close this PR?

**Comment by dalssoft** [comment]
@italojs agree. I'm closing it. Tks

**Pull request closed without merging**